account_financial_report/report/trial_balance.py

```diff
@@ -226,21 +226,37 @@ def _compute_account_amount(
                     )
         return total_amount
 
+    @api.model
+    def _prepare_total_amount(self, foreign_currency):
+        res = {
+            "credit": 0.0,
+            "debit": 0.0,
+            "balance": 0.0,
+            "initial_balance": 0.0,
+            "ending_balance": 0.0,
+        }
+        if foreign_currency:
+            res["initial_currency_balance"] = 0.0
+            res["ending_currency_balance"] = 0.0
+        return res
+
     @api.model
     def _compute_acc_prt_amount(
         self, total_amount, tb, acc_id, prt_id, foreign_currency
     ):
-        total_amount[acc_id][prt_id] = {}
-        total_amount[acc_id][prt_id]["credit"] = 0.0
-        total_amount[acc_id][prt_id]["debit"] = 0.0
-        total_amount[acc_id][prt_id]["balance"] = 0.0
-        total_amount[acc_id][prt_id]["initial_balance"] = tb["balance"]
-        total_amount[acc_id][prt_id]["ending_balance"] = tb["balance"]
+        # Add keys to dict if not exists
+        if acc_id not in total_amount.keys():
```

Comment:
The `in` expression over a `dict` already check if the object is a key.
```suggestion
        if acc_id not in total_amount:
```